java/debugger.jpda/src/org/netbeans/modules/debugger/jpda/models/ShortenedStrings.java

```diff
@@ -92,6 +108,73 @@ public static StringInfo getShortenedInfo(String s) {
         }
     }
 
+    private static boolean isLittleEndian(VirtualMachine virtualMachine) {
+        synchronized(isLittleEndianCache){
+            Boolean cached = isLittleEndianCache.get(virtualMachine);
+            if (cached != null){
+                return cached;
+            }
+            List<ReferenceType> possibleClasses = virtualMachine.classesByName(
+                    "java.lang.StringUTF16");
+            //If we don't know, we are going to assume little endian encoding.
+            //This should work for most architectures (x86, arm, riscv), but
+            //will result in bogus data on big endian architectures
+            final boolean defaultValue = true;
+            if (possibleClasses.isEmpty()){
+                ClassType ct = (ClassType) virtualMachine.classesByName(
+                        "java.lang.Class").iterator().next();
+                Method m = ct.concreteMethodByName("forName",
+                        "(Ljava/lang/String;)Ljava/lang/Class;");
+                StringReference referenceString = virtualMachine.mirrorOf(
+                        "java.lang.StringUTF16");
+                try {
+                    ThreadReference threadReference = virtualMachine.
+                            allThreads().get(0);
+                    ct.invokeMethod(threadReference, m, Collections.
+                            singletonList(referenceString), 0);
+                    possibleClasses = virtualMachine.classesByName(
+                            "java.lang.StringUTF16");
+                } catch (InvalidTypeException | ClassNotLoadedException |
+                        IncompatibleThreadStateException | InvocationException ex) {
+                    Exceptions.printStackTrace(ex);
+                    isLittleEndianCache.put(virtualMachine, defaultValue);
+                    return defaultValue;
+                }
+            }
+            ReferenceType utf16;
+            if (possibleClasses.size() == 1){
+                utf16 = possibleClasses.get(0);
+            }
+            else {
+                isLittleEndianCache.put(virtualMachine, defaultValue);
+                return defaultValue;
+            }
+            Field hiByteShift;
```

Comment:
Please follow the naming convention you also used in the other method and make this `hiByteShiftField`